### PR TITLE
refactor: Interface updates

### PR DESCRIPTION
**Pull request by 0xJurassicPunk**
This PR:
- Refactors interfaces with event/errors outside of the implementation code
- Add other generic interfaces (e.g., IERC20, IERC721)

Note: It excludes the `OwnableTwoSteps` contract (since it is refactored in another PR --> https://github.com/LooksRare/contracts-libs/pull/20).

### CODE REVIEW

File: test/foundry/ReentrancyGuard.t.sol [hunk 1]
```diff
@@ -79,11 +79,7 @@ contract ReentrancyCaller {
     }
 }
 
-abstract contract ReentrancyGuardErrors {
-    error ReentrancyFail();
-}
-
-contract ReentrancyGuardTest is TestHelpers, ReentrancyGuardErrors {
+contract ReentrancyGuardTest is TestHelpers {
```
Comment (0xhiroshi) [inline test/foundry/ReentrancyGuard.t.sol:82]: Is this going to inherit from `IReetrancyGuard`?

Comment (0xJurassicPunk) [reply test/foundry/ReentrancyGuard.t.sol:82]: Actually, it reverts on the fallback and I could not catch the reversion with `vm.expectRevert()` (although I see, as expected, the error `ReentrancyFail` in the stack traces). That's why I didn't import the interface.

https://github.com/LooksRare/contracts-libs/blob/165c6e9babf6ed27cafef7d038bdf29ae350706d/test/foundry/ReentrancyGuard.t.sol#L106

Comment (0xhiroshi) [reply test/foundry/ReentrancyGuard.t.sol:82]: Try this in SafeFaucet, you have an empty `revert()` so the `ReentrancyFail` is not bubbling up, failing with `EvmError: revert` instead

```
assembly {
    status := call(gas(), to, amount, 0, 0, 0, 0)
    // returndatacopy(t, f, s)
    // copy s bytes from returndata at position f to mem at position t
    returndatacopy(0, 0, returndatasize())
    switch status
    case 0 {
        // revert(p, s)
        // end execution, revert state changes, return data mem[p…(p+s))
        revert(0, returndatasize())
    }
}
```

Comment (0xJurassicPunk) [reply test/foundry/ReentrancyGuard.t.sol:82]: Nice, it worked well. Let me adjust.

**Review by 0xhiroshi — APPROVED** [review]

**Pull request merged**